conf/modules.config

```diff
@@ -8,21 +8,23 @@
         ext.args3           = Third set of arguments appended to command in module (multi-tool modules).
         ext.suffix          = File name ext.suffix output files. Not available for nf-core modules
         ext.prefix          = File name prefix for output files.
+        ext.when            = When to run the module.
 ----------------------------------------------------------------------------------------
 */
 
 process {
 
-    publishDir = [
-        path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
-        mode: 'copy',
-        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
-    ]
+    withName: '.*' {
+        publishDir = [
+            path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+            mode: "${params.publish_dir_mode}",
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
 
```

Comment:
this seems to work for all processes :-D
Might need to do more tests, but at least we can get rid of this `mode: 'copy'` everywhere

Reply:
awesome

Reply:
and it doesn't get overwritten if you set a different path or so?

Reply:
good point, I'll look that up, thanks for the idea.
Maybe I can ping @nf-core/core and @mahesh-panchal here actually

Reply:
It doesn't inherit. Toy:
`main.nf`:
```nextflow
#! /usr/bin/env nextflow

nextflow.enable.dsl = 2

workflow {

    FOO()
}

process FOO {

    output:
    path "output.txt"

    script:
    """
    echo "Message" > output.txt
    """
}
```
`nextflow.config`:
```nextflow
params {
    outdir = 'results'
    publish_dir_mode = 'symlink'
}
process {
    withName: '.*' {
        publishDir = [
            path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
            mode: "${params.publish_dir_mode}",
            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
        ]
    }
    withName: 'FOO' {
        publishDir = [
            path: { "${params.outdir}/FOO" },
            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
        ]
    }
}
```
output:
```console
$ nextflow run main.nf 
N E X T F L O W  ~  version 21.10.6
Launching `main.nf` [mad_bell] - revision: 9e7d761695
executor >  local (1)
[de/9fcfaf] process > FOO [100%] 1 of 1 ✔

$ tree results/
results/
└── FOO
    └── output.txt -> /Users/mahpa906/Downloads/Nextflow_sandbox/test_publishDir_inheritance/work/de/9fcfaf5a17a2d96fb161fb364a693c/output.txt

1 directory, 1 file
$ nextflow run main.nf --publish_dir_mode 'copy'
N E X T F L O W  ~  version 21.10.6
Launching `main.nf` [silly_coulomb] - revision: 9e7d761695
executor >  local (1)
[32/e33258] process > FOO [100%] 1 of 1 ✔

$ tree results/
results/
└── FOO
    └── output.txt -> /Users/mahpa906/Downloads/Nextflow_sandbox/test_publishDir_inheritance/work/32/e3325849ffb8e3f2cc0e2e0fde833e/output.txt

1 directory, 1 file
```

Reply:
do you think it can work with a different `withName` selector?

Reply:
What do you mean?

Reply:
So far all the toys I've tried, needed to provide `mode: params.publish_dir_mode` when overriding. There's no form of inheritance.

Reply:
Did I misread your test?
Is it working or not?

Reply:
ok, so still need to keep this everywhere then...

Reply:
😢 oh well